WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewController.swift

```diff
@@ -111,13 +124,19 @@ private extension ShippingLabelAddressFormViewController {
 private extension ShippingLabelAddressFormViewController {
 
     @objc func doneButtonTapped() {
-        // TODO: after the validation, return to the previous controller passing the new address.
+        configureRightButtonItemAsLoader()
```

Comment:
Tip:
If you move this logic into the VM like:

https://github.com/woocommerce/woocommerce-ios/blob/dd84c1d3064f8c065cc293079619ccc399a6d0d9/WooCommerce/Classes/ViewRelated/Products/Variations/Add%20Attributes/AddAttributeOptionsViewModel.swift#L84-L88

You can add unit tests to corroborate this behavior! 

https://github.com/woocommerce/woocommerce-ios/blob/dd84c1d3064f8c065cc293079619ccc399a6d0d9/WooCommerce/WooCommerceTests/ViewRelated/Variations/Add%20Attributes/AddAttributeOptionsViewModelTests.swift#L340-L363

Reply:
Do you mean to introduce a state variable for managing the various states (in this moment the states are just two: `isUpdating` and `loaded`)?

Reply:
Yes, adding that state into the VM so it can be tested!

Reply:
The state can be stored into a single var,  `private var: isUpdating: Bool`
And then 
```swift
var showUpdateIndicator: Bool { 
    state.isUpdating 
} 
```

Reply:
Done 5c2a64c94c7a289e2e4f853e289c77657ef33164 👍